File: data-structures/stack/python/stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, value):
        self.stack.append(value)

    def pop(self):
        if not self.stack:
            return None

        return self.stack.pop()

    def size(self):
        return len(self.stack)

    def peek(self):
        if not self.stack:
            return None

        return self.stack[-1]

    def print_stack(self):
        return self.stack

File: data-structures/stack/typescript/stack.ts
export class Stack<T> {
	private stack: T[] = [];
	constructor() {}

	push(value: T) {
		this.stack.push(value);
	}

	pop(): T | null {
		if (!this.stack.length) {
			return null;
		}
		return this.stack.pop()!;
	}

	size(): number {
		return this.stack.length;
	}

	peek(): T | null {
		if (!this.stack.length) {
			return null;
		}
		return this.stack[this.stack.length - 1];
	}

	printStack() {
		return this.stack;
	}
}
